tests/cypress/e2e/api/jcr/graphqlReferences.cy.ts

```diff
@@ -0,0 +1,250 @@
+            children: [
+                {
+                    name: 'testSubList1',
+                    primaryNodeType: 'jnt:contentList',
+                    properties: [{name: 'jcr:title', value: '/testList/testSubList2', language: 'en'}]
+                },
+                {
+                    name: 'testSubList2',
+                    primaryNodeType: 'jnt:contentList',
+                    properties: [{name: 'jcr:title', value: '/nonExistingPath', language: 'en'}],
+                    mixins: ['jmix:unstructured']
+                },
+                {
+                    name: 'reference1',
+                    primaryNodeType: 'jnt:contentReference'
+                },
+                {
+                    name: 'reference2',
+                    primaryNodeType: 'jnt:contentReference'
+                }
+            ]
+        }).then(response => {
+            const subNode1Uuid = response.data.jcr.addNode.addChildrenBatch[0].uuid;
+            setNodeProperty('/testList/reference1', 'j:node', subNode1Uuid, 'en');
+            setNodeProperty('/testList/reference2', 'j:node', subNode1Uuid, 'en');
+            addNode({
+                parentPathOrId: '/testList/testSubList2',
+                primaryNodeType: 'nt:linkedFile',
+                name: 'reference3',
+                properties: [{name: 'jcr:content', value: subNode1Uuid}]
+            });
+        });
+    });
+
+    it('should retrieve references', () => {
+        cy.apollo({
+            query: gql`
+                query {
+                    jcr {
+                        nodeByPath(path: "/testList/testSubList1") {
+                            references {
+                                nodes {
+                                    node {
+                                        name
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            `
+        }).should(result => {
+            const references = result?.data?.jcr?.nodeByPath?.references?.nodes as Array<{node: {name: string}}>;
+            const referenceNames = references.map(r => r.node.name);
+            expect(referenceNames).to.contain('reference1');
+            expect(referenceNames).to.contain('reference2');
+            expect(referenceNames).to.contain('reference3');
+        });
+    });
+
+    it('should retrieve referenced node by reference', () => {
+        cy.apollo({
+            query: gql`
+                query {
+                    jcr {
+                        nodeByPath(path: "/testList/reference1") {
+                            property(name: "j:node") {
+                                refNode {
+                                    name
+                                }
+                            }
+                        }
+                    }
+                }
+            `
```

Comment:
There's a bit of repetition of this query with the rest of the test functions. Maybe appropriate to refactor this graphql as a fixture (tests/cypress/fixture) with parameters.